internal/bufferedread/buffered_reader.go

```diff
@@ -98,6 +98,44 @@ func NewBufferedReader(object *gcs.MinObject, bucket gcs.Bucket, config *Buffere
 	return reader, nil
 }
 
+// ScheduleNextBlock schedules the next block for prefetch.
+func (p *BufferedReader) ScheduleNextBlock(urgent bool) error {
+	b, err := p.blockPool.Get()
```

Comment:
Add a TODO(bug_id): to replace it with TryGet(). Assuming, the current blockPool.Get() gets blocked if block not available.